### PR TITLE
Some cleaning up

### PR DESCRIPTION
**Pull request by JonathanWoollett-Light**
No change to any functionality.

Primarily:

1. Ran `cargo fmt`
2. Fixed spelling and grammar errors.
3. Added rustdoc links and some small edits.

### CODE REVIEW

**Comment by nestordemeure** [comment]
For `cargo fmt`, could you use that file (and include it in the repository, I thought I did): https://github.com/nestordemeure/Simplers/blob/master/rustfmt.toml

**Comment by JonathanWoollett-Light** [comment]
Just added the custom fmt.

File: src/gaussian_process/mod.rs [hunk 1]
```diff
@@ -183,9 +180,9 @@ impl<KernelType: Kernel, PriorType: Prior> GaussianProcess<KernelType, PriorType
     {
         // formula : -1/2 (transpose(output)*cov(train,train)^-1*output + trace(log|cov(train,train)|) + size(train)*log(2*pi))
 
-        // how well do we fit the trainnig data ?
+        // How well do we fit the training data?
         let output = self.training_outputs.as_vector();
-        // transpose(ol)*ol = transpose(output)*cov(train,train)^-1*output
+        // Transpose(ol)*ol = transpose(output)*cov(train,train)^-1*output
```
Comment (nestordemeure) [inline src/gaussian_process/mod.rs:185]: I would keep this transpose uncapitalised but its a minor thing.

**Comment by nestordemeure** [comment]
Thank you for the cleaning-up work, that is a lot of great work!

**Pull request merged**